authentication/auth_method_oidc.go

```diff
@@ -32,7 +35,7 @@ func (a oidcAuth) build(b Builder) (authMethod, error) {
 }
 
 func (a oidcAuth) isApplicable(b Builder) bool {
-	return b.SupportsOIDCAuth && b.IDTokenRequestURL != "" && b.IDTokenRequestToken != "" && b.UseMicrosoftGraph
+	return b.SupportsOIDCAuth && (b.IDToken != "" || (b.IDTokenRequestURL != "" && b.IDTokenRequestToken != "")) && b.UseMicrosoftGraph
```

Comment:
Just a small suggestion to make this slightly more readable

```suggestion
	return b.SupportsOIDCAuth && b.UseMicrosoftGraph && (b.IDToken != "" || (b.IDTokenRequestURL != "" && b.IDTokenRequestToken != ""))
```